app/models/detailed_guide.rb

```diff
@@ -145,18 +138,54 @@ def published_inbound_related_detailed_guides
     DetailedGuide.published.joins(:outbound_edition_relations).where(edition_relations: { document_id: document.id })
   end
 
-  def related_mainstream_content_valid?
-    if related_mainstream_content_url.present? && related_mainstream_content_title.blank?
-      errors.add(:related_mainstream_content_title, "cannot be blank if a related URL is given")
+  def related_mainstream_found
+    return unless related_mainstream_requested?
+    fetch_related_mainstream_content_ids
+    add_errors_for_missing_related_mainstream
+    add_errors_for_missing_additional_related_mainstream
+  end
+
+  def add_errors_for_missing_related_mainstream
+    if @content_ids.count >= 1
+      errors.add(:related_mainstream_content_url, "This mainstream content could not be found") if @content_ids[0].nil?
     end
   end
 
-  def additional_related_mainstream_content_valid?
-    if additional_related_mainstream_content_url.present? && additional_related_mainstream_content_title.blank?
-      errors.add(:additional_related_mainstream_content_title, "cannot be blank if an additional related URL is given")
+  def add_errors_for_missing_additional_related_mainstream
+    if @content_ids.count > 1
+      errors.add(:additional_related_mainstream_content_url, "This mainstream content could not be found") if @content_ids[1].nil?
     end
   end
 
+  def fetch_related_mainstream_content_ids
+    base_paths = [related_mainstream_base_path, additional_related_mainstream_base_path].compact
+    if base_paths.any?
+      @content_ids ||= lookup_content_ids(base_paths)
+    else
+      @content_ids ||= []
+    end
+  end
+
+  def lookup_content_ids(base_paths)
+    @content_ids = []
+    base_paths.each do |base_path|
```

Comment:
In that case, you can use the hash it returns:
```ruby
irb(main):009:0> Whitehall.publishing_api_v2_client.lookup_content_ids(base_paths: %w{/government /foobar})
=> {"/government"=>"4672b1ff-f147-4d49-a5f4-4959588da5a8"}
```
It's definitely better to make one call than two.